source/Utils/src/FilterTracks.cc

```diff
@@ -132,6 +144,19 @@ void FilterTracks::processEvent( LCEvent * evt )
 
     float chi2spatial = trk->getChi2();
 
+    int nholes = trk->getNholes();
+
+    bool foundCaloState = false;
+    // Check if a TrackState at the calo surface exists
+    const std::vector<EVENT::TrackState*>& trackStates = trk->getTrackStates();
+    for (const auto& state : trackStates) {
+      if (state->getLocation() == EVENT::TrackState::AtCalorimeter) {
+        foundCaloState = true;
+        break;
+      }
+    }
```

Comment:
```suggestion
    const auto foundCaloState = std::find_if(trackStates.begin(), trackStates.end(), 
                                             [](const auto ts) { return ts->getLocation() == EVENT::TrackState::AtCalorimeter) 
                                != trackStates.end();
```
(or if we can / want to break c++17 compatibilty, we could also use `std::ranges::find_if(trackStates, [](const auto ts) { /* as above */ } != trackStates.end();`

This would make it pretty clear what is being done, without having to "decipher" the loop first.